### ISSUE TITLE
Filter placeholder and numeric sorting

### ISSUE THREAD

**Issue by ronwillemse** [GriddleGriddle/Griddle#774]
## Griddle version
1.9.0

Hi, is there an option to change the filter placeholdertext to something like search without writing a custom filter component? I think it is much overdone to add a custom component only to change the placeholder text.

Second thing is the way of sorting. For example if i have numbers it sorts like:

1
10
100
1000
2
3

instead of

1
2
3
10
...

How can I solve this?

**Comment by dahlbyk** [comment]
> is there an option to change the filter placeholdertext to something like search without writing a custom filter component? I think it is much overdone to add a custom component only to change the placeholder text.

Not currently, but this should be a relatively easy patch if you want to submit a PR:

1. Add `filterPlaceholder` to [`initialState.textProperties`](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/src/core/initialState.js#L40-L44)
2. Use `textSelector` in `FilterContainer` to add that value to `props` (e.g. [`PreviousButtonContainer`](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/src/components/PreviousButtonContainer.js#L6)
3. Use the new `props` value in `Filter`

Once connected to `textProperties`, you can override the text [from a plugin](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/stories/index.tsx#L1679-L1684) or [directly](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/stories/index.tsx#L1535).

> Second thing is the way of sorting. For example if i have numbers it sorts like...

You can provide a custom `sortMethod` either for the entire grid or specific columns. Check out how [this custom sort function](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/stories/index.tsx#L27-L42) is used in the Storybook.

**Comment by ronwillemse** [comment]
If it is possible to change the textproperties for next and previous than it should also be possible to change other values like the placeholder. Let me try to find a workaround with te use of textSelector

**Comment by dahlbyk** [comment]
I agree that it should be possible. And the Filter placeholder isn't the only example of hard-coded text that could be converted to use `textSelector` (e.g. [`NoResults`](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/src/components/NoResults.js#L5), [`Loading`](https://github.com/GriddleGriddle/Griddle/blob/1e6346ab37df96adcd69a22ef2959f7be2f5db00/src/components/Loading.js#L5)).

If you're not interested in contributing, we'll try to fix this as time allows.

**Comment by miguelsaldivar** [comment]
I made a pr for the first issue 😸

**Comment by ronwillemse** [comment]
I'm curious and I will check it out
